account_invoice_merge/models/account_move.py

```diff
@@ -0,0 +1,255 @@
+# Copyright 2004-2010 Tiny SPRL (http://tiny.be).
+# Copyright 2010-2011 Elico Corp.
+# Copyright 2016 Acsone (https://www.acsone.eu/)
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2019 Okia SPRL
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import numbers
+
+from odoo import Command, api, models
+from odoo.tools import float_is_zero
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def _get_invoice_key_cols(self):
+        return [
+            "partner_id",
+            "user_id",
+            "move_type",
+            "currency_id",
+            "journal_id",
+            "company_id",
+            "partner_bank_id",
+        ]
+
+    @api.model
+    def _get_invoice_line_key_cols(self):
+        fields = [
+            "name",
+            "discount",
+            "tax_ids",
+            "price_unit",
+            "product_id",
+            "account_id",
+            "analytic_distribution",
+            "product_uom_id",
+        ]
+        for field in [
+            "sale_line_ids",  # odoo/sale
+            "purchase_line_id",  # odoo/purchase
+            "purchase_price",  # OCA/account_invoice_margin
+        ]:
+            if field in self.env["account.move.line"]._fields:
+                fields.append(field)
+        return fields
+
+    @api.model
+    def _get_first_invoice_fields(self, invoice):
+        return {
+            "invoice_origin": f"{invoice.invoice_origin or ''}",
```

Comment:
f-string has no use here
```suggestion
            "invoice_origin": invoice.invoice_origin or "",
```